### ISSUE TITLE
RequestLayout() improperly called

### ISSUE THREAD

**Issue by A-Lim** [roughike/BottomBar#371]
I've got these warnings whenever the Tab Bar activity is launched. 

`W/View: requestLayout() improperly called by android.support.v7.widget.AppCompatTextView{2af626cd V.ED.... ......ID 109,111-215,168 #7f0e0075 app:id/bb_bottom_bar_title} during layout: running second layout pass
W/View: requestLayout() improperly called by android.support.v7.widget.AppCompatTextView{1a2f2b82 V.ED.... ......ID 32,120-161,177 #7f0e0075 app:id/bb_bottom_bar_title} during layout: running second layout pass
W/View: requestLayout() improperly called by android.support.v7.widget.AppCompatTextView{2de35293 V.ED.... ......ID 9,120-185,177 #7f0e0075 app:id/bb_bottom_bar_title} during layout: running second layout pass
W/View: requestLayout() improperly called by android.support.v7.widget.AppCompatTextView{3da522d0 V.ED.... ......ID 0,120-194,177 #7f0e0075 app:id/bb_bottom_bar_title} during layout: running second layout pass
W/View: requestLayout() improperly called by android.support.v7.widget.AppCompatTextView{ccb62c9 V.ED.... ......ID 37,120-157,177 #7f0e0075 app:id/bb_bottom_bar_title} during layout: running second layout pass`

Any idea on what might be the problem?

**Comment by jessexx** [comment]
I have the same issue, it doesn't seem to be affecting performance in any way but it shows up in the logger. I'm attaching the BottomBar with five items that have Drawables for icons if that helps.

**Comment by jchen236** [comment]
I am also getting these warnings.

**Comment by mdeyell** [comment]
I am also getting these, with 5 icons.

**Comment by JEKos** [comment]
Me too

**Comment by REWKyleB** [comment]
+1

**Comment by firetrap** [comment]
+1 and the textView it's being cut on every click

**Comment by UmarBhutta** [comment]
+1 and text is not showing on Android L

**Comment by a00909032** [comment]
trying to figure this out too... any solution yet?

**Comment by Rikkokiri** [comment]
Yeah I have the same issue.

**Comment by iRYO400** [comment]
thumb up! Still in Ver. 2.1.1

**Comment by strajky5** [comment]
i have the same issue with ver. 2.1.1

**Comment by Zoomagic** [comment]
+1

**Comment by heitorpr** [comment]
+1 the same here with the same scenario. BottomBar with 5 itens all with icons and labels.

**Comment by chaleao** [comment]
+1

**Comment by burhanaksendir** [comment]
I have the same issue.

**Comment by kandreadou** [comment]
Same issue here

**Comment by yombunker** [comment]
is this happening on the latest version? 2.2.0

**Comment by roughike** [comment]
Confirmed on the latest dev branch, looking into it.

**Comment by yombunker** [comment]
@roughike this is because we trigger a layout phase while on the layout phase, i think what we are missing is that we are not leveraging the measuring phase to calculate the sizes and just lay them out on the payout phase.

**Comment by roughike** [comment]
Yep. There was no manual ```requestLayout``` method calls that were out of place, so I guess the system calls that for us whenever we update the width / height, paddings, etc. while the onMeasure method hasn't finished. 

There's a lot of optimizations to be done through the library. The weird ```updateTitleBottomPadding()``` method for example should be absolutely unnecessary. It was introduced when merging some PR like almost a year ago and to this day, I don't know why it needs to be here at all.

**Comment by burhanaksendir** [comment]
https://github.com/roughike/BottomBar/blob/master/bottom-bar/src/main/java/com/roughike/bottombar/BottomBarBadge.java#L32

I resolved this using android.support.v7.widget.AppCompatTextView:

class BottomBarBadge extends android.support.v7.widget.AppCompatTextView {

**Comment by yombunker** [comment]
@burhanaksendir if that's teh fix, then it will be coming in the next days because that's part of the change i'm working on

**Comment by nextdimension** [comment]
Its still bugged, any eta on fix?

**Comment by JDDJJ** [comment]
when will this bug be fixed? this bug will cause UI stop when app open for long time.

**Comment by nextdimension** [comment]
Its been 1 year and no one fixed it, I'd say they don't know how to, or can't. I'll just use a different bottom bar.

**Comment by yombunker** [comment]
@nextdimension @JDDJJ well this library wasn't maintained for a while, And i just started helping about like 2 months ago, but, this is just like a side thing, i'm like 60% done with a huge refactoring in how the BottomBar works, which would make it better while still being backwards compatible. But, i'm super swamped with work this weeks that i haven't been able to finish it. But, don't worry @nextdimension i know how to do it, is already solved, the missing thing is not that is other things before the next release is ready. But, you are always free to use other libraries, or if you want you can contribute Pull Request to improve it as well, i try to take a look to PR as they arrive.

**Comment by JDDJJ** [comment]
![beced8ea-4db3-4b4b-8f32-e6dcd2b6d2f9](https://cloud.githubusercontent.com/assets/16041527/26055448/9dff8a76-39a3-11e7-83b6-1c56808fcdca.png)
![e7b71f25-82e3-4384-a301-16e6048eed92](https://cloud.githubusercontent.com/assets/16041527/26055449/9e02cd94-39a3-11e7-838f-750fcfec4a08.png)

**Comment by JDDJJ** [comment]
![5a621999-9628-488a-af9f-1816cf79b9ee](https://cloud.githubusercontent.com/assets/16041527/26055583/1f9880f6-39a4-11e7-865e-724347163909.png)
![Uploading BECED8EA-4DB3-4B4B-8F32-E6DCD2B6D2F9.png…]()

**Comment by JDDJJ** [comment]
![5a621999-9628-488a-af9f-1816cf79b9ee](https://cloud.githubusercontent.com/assets/16041527/26055611/35aef3ac-39a4-11e7-8813-4f06aab56c3e.png)
![beced8ea-4db3-4b4b-8f32-e6dcd2b6d2f9](https://cloud.githubusercontent.com/assets/16041527/26055612/35b0896a-39a4-11e7-8ac0-4cc9ef3a2566.png)

**Comment by massimobio** [comment]
I tried the fix recommended @burhanaksendir but the warnings and the 1-2sec lag on rotation remains

**Comment by Albul** [comment]
Why do you need this updateTitleBottomPadding ?
What if just add to bb_bottom_bar_item_shifting.xml 
<TextView
.....
        android:paddingBottom="10dp"
..../>

**Comment by Albul** [comment]
Fixed in pull request
https://github.com/roughike/BottomBar/pull/866

Working fine in my project.
